### DISCUSSION TITLE
Issue with RandomLinkSplit in train-val split for Heterogeneous Graph

### DISCUSSION THREAD

**Discussion by aaronwtr** [pyg-team/pytorch_geometric#8784] [Q&A]
Hi everyone,

I'm encountering unexpected behavior with the RandomLinkSplit in PyTorch Geometric. Before splitting edges, my original graph has 236,448 edges. I'm using the following code for the train-test split:

```py
def train_test_split(graph):
    edge_type_list = list(graph.edge_types)
    rev_edge_type_list = edge_type_list.copy()
    transform = T.RandomLinkSplit(
        num_val=0.15,
        num_test=False,
        is_undirected=True,
        key="target",
        add_negative_train_samples=False,
        edge_types=edge_type_list,
        rev_edge_types=rev_edge_type_list
    )
    return transform(graph)
```
then calling 
```py
train_graph, val_graph, test_graph = random_link_split(graph)
```

Despite setting num_test to False, the resulting graphs have unexpected edge counts:

- train_graph.num_edges = 200986
- val_graph.num_edges = 200986
- test_graph.num_edges = 236452

Two peculiarities: 1) the test-to-val ratio is not 0.15 as specified, and 2) the test graph contains edges despite explicitly setting num_test to False. The same thing happens if I set num_test = 0.0

Also, if I set num_test = 0.1 and num_val =0.1, I get 

- train_graph.num_edges = 189168
- val_graph.num_edges = 189168
- test_graph.num_edges = 212810

I would greatly appreciate any insights into what might be causing this behavior. Thanks in advance for your help!

**Comment by rusty1s** [comment]
I think you mix up `edge_index` and `edge_label_index`. `edge_label_index` should indeed have a ratio of `0.85/0.15/0`. `edge_index` just denotes the additional edges used for message passing.

**Reply by aaronwtr** [reply]
I think that makes sense - we are not separating _edges_ in train-test splitting, but rather the _edge labels_? I suppose I got confused by the docs in RandomLinkSplit which states: 

"Performs an edge-level random split into training, validation and test sets of a [Data](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.data.Data.html#torch_geometric.data.Data) or a [HeteroData](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.data.HeteroData.html#torch_geometric.data.HeteroData) object (functional name: random_link_split). The split is performed such that the training split does not include edges in validation and test splits; and the validation split does not include edges in the test split."

In this explanation, should it then be _edge labels_ instead of _edges_?

**Reply by rusty1s** [reply]
I think the doc string is fine, meaning that no supervision edges of validation and test split are visible for message passing during training; and no supervision edges of the test split are visible for message passing during validation.
